### ISSUE TITLE
Is timeseries(RNN) analysis possible with shap

### ISSUE THREAD

**Issue by arpanaalka** [shap/shap#444]
Is it possible to analyze with shap for timeseries  where 2 dimensional  input is used

**Comment by slundberg** [comment]
Yes. GradientExplainer and DeepExplainer both directly explain tensors of any order. TreeExplainer and KernelExplainer would require you to reshape each sample into a single row.

**Comment by shamitb** [comment]
Can SHAP be used with models such as VARMAX etc?

**Comment by slundberg** [comment]
@shamitb yes, but until your varmax model is in tensorflow or Pytorch then you would have to use one of the model agnostic explainers (like KernelExplainer). Since VARMAX is linear you could also just write out the SHAP values using the closed form solution for linear models: `coef[j] * (X[i,j] - X.mean(0)[j])`.

**Comment by abis330** [comment]
Is this issue resolved now? I have a similar type of use case where I want to interpret an LSTM-model predicting a continuous value in future time steps?

**Comment by ArpanaAlkaQuantela** [comment]
> Is this issue resolved now? I have a similar type of use case where I want to interpret an LSTM-model predicting a continuous value in future time steps?

yes, I have got the values as per my use case

**Comment by shubhij** [comment]
> > Is this issue resolved now? I have a similar type of use case where I want to interpret an LSTM-model predicting a continuous value in future time steps?
> 
> yes, I have got the values as per my use case

I am facing this issue in using each time stamp as a row:
(My sample size is 3031 and my max_len is 400)

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-132-96f1fe081f51> in <module>
----> 1 explainer = shap.KernelExplainer(predictions, t3)

/axp/platform/mlplatml/app/mlstudio/python37/envs/mlsi_1_0_2021_07_06/lib/python3.7/site-packages/shap/explainers/kernel.py in __init__(self, model, data, link, **kwargs)
    118         if isinstance(model_null, (pd.DataFrame, pd.Series)):
    119             model_null = np.squeeze(model_null.values)
--> 120         self.fnull = np.sum((model_null.T * self.data.weights).T, 0)
    121         self.expected_value = self.linkfv(self.fnull)
    122 

ValueError: operands could not be broadcast together with shapes (1,3031) (1212400,)

**Comment by shubhij** [comment]
I have built my RNN model on sequential data. Now I am trying to transform each time stamp as a row to explain the predictions but it is throwing error related to dimensions of arrays. Is there any other way to explain predictions for a time series data?

**Comment by pedro-zenario** [comment]
> I have built my RNN model on sequential data. Now I am trying to transform each time stamp as a row to explain the predictions but it is throwing error related to dimensions of arrays. Is there any other way to explain predictions for a time series data?

Were you able to solve this?
